lib/bundler/compatibility_guard.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: false
+
+require "rubygems"
+require "bundler/version"
+
+if Bundler::VERSION.split(".").first.to_i >= 2
+  if Gem::Version.new(Object::RUBY_VERSION.dup) < Gem::Version.new("2.3")
+    abort "Bundler 2 requires Ruby 2.3 or later. Either install bundler 1 or update to a supported Ruby version."
```

Comment:
Suggest a less terse statement and flip the action order to show our first choice is upgrade ruby and the fallback plan is to go with v1.x.
`Bundler 2.x requires Ruby 2.3 or later. Please update your Ruby version to 2.3 or higher, or use the latest Bundler 1.x.`